pkg/cvo/metrics.go

```diff
@@ -129,17 +172,35 @@ func RunMetrics(runContext context.Context, shutdownContext context.Context, lis
 	resultChannel := make(chan asyncResult, 1)
 	resultChannelCount := 1
 
-	go func() {
-		tlsListener := tls.NewListener(tcpListener, tlsConfig)
-		klog.Infof("Metrics port listening for HTTPS on %v", listenAddress)
-		err := server.Serve(tlsListener)
-		resultChannel <- asyncResult{name: "HTTPS server", error: err}
-	}()
+	go startServing(server, tcpListener, tlsConfig, listenAddress, resultChannel)
+
+	certDir := filepath.Dir(certFile)
+	keyDir := filepath.Dir(keyFile)
+	origCertChecksum := checksumFile(certFile)
+	origKeyChecksum := checksumFile(keyFile)
+
+	// Set up and start the file watcher.
+	watcher, err := fsnotify.NewWatcher()
+	if watcher == nil || err != nil {
+		return fmt.Errorf("Failed to create file watcher for certificate and key rotation: %v", err)
```

Comment:
nit: [`%w` is like `%v` but includes `Unwrap` support][1].  Can we pivot these new entries to that?

[1]: https://pkg.go.dev/fmt#Errorf